### DISCUSSION TITLE
Upsert Bulk

### DISCUSSION THREAD

**Discussion by hojabbr** [jenssegers/laravel-mongodb#1415] [General] [answered]
I'm trying to do an upsert with bulk but I get duplicate rows:

`DB::table('items')->raw(function ($collection) use ($channels) {
            return $collection->insertMany($channels, ['upsert' => true]);
        })`

$channels is an array of at least 500 items:

`                $channels[] = [
                    'title' => $username,
                    'slug' => $username,
                    'active' => 0,
                    'type_id' => 1
                ];`

**Comment by GromNaN** [comment, marked as answer]
This is supported using the `upsert` method ([doc](https://www.mongodb.com/docs/drivers/php/laravel-mongodb/v4.x/fundamentals/write-operations/#update-or-insert-in-a-single-operation)).
